internal/services/network/application_gateway_resource.go

```diff
@@ -609,9 +753,11 @@ func resourceApplicationGateway() *pluginsdk.Resource {
 			},
 
 			"request_routing_rule": {
-				Type:     pluginsdk.TypeSet,
-				Required: true,
-				MinItems: 1,
+				Type:         pluginsdk.TypeSet,
+				Optional:     true,
+				MinItems:     1,
+				RequiredWith: []string{"backend_http_settings", "http_listener"},
+				AtLeastOneOf: []string{"request_routing_rule", "routing_rule"},
```

Comment:
Do you mean `ExactlyOneOf`?

Reply:
No, they can both exists. But at least one of them needs exist.